ceremonies/src/lib.rs

```diff
@@ -766,37 +773,41 @@ impl<T: Config> Pallet<T> {
 			<NewbieIndex<T>>::contains_key((cid, cindex), &sender)
 	}
 
-	fn purge_registry(cindex: CeremonyIndexType) {
-		let cids = <encointer_communities::Pallet<T>>::community_identifiers();
-		for cid in cids.iter() {
-			<BootstrapperRegistry<T>>::remove_prefix((cid, cindex), None);
-			<BootstrapperIndex<T>>::remove_prefix((cid, cindex), None);
-			<BootstrapperCount<T>>::insert((cid, cindex), 0);
+	fn purge_community_ceremony(cc: CommunityCeremony) {
+		<BootstrapperRegistry<T>>::remove_prefix(cc, None);
+		<BootstrapperIndex<T>>::remove_prefix(cc, None);
+		<BootstrapperCount<T>>::insert(cc, 0);
 
-			<ReputableRegistry<T>>::remove_prefix((cid, cindex), None);
-			<ReputableIndex<T>>::remove_prefix((cid, cindex), None);
-			<ReputableCount<T>>::insert((cid, cindex), 0);
+		<ReputableRegistry<T>>::remove_prefix(cc, None);
+		<ReputableIndex<T>>::remove_prefix(cc, None);
+		<ReputableCount<T>>::insert(cc, 0);
 
-			<EndorseeRegistry<T>>::remove_prefix((cid, cindex), None);
-			<EndorseeIndex<T>>::remove_prefix((cid, cindex), None);
-			<EndorseeCount<T>>::insert((cid, cindex), 0);
+		<EndorseeRegistry<T>>::remove_prefix(cc, None);
+		<EndorseeIndex<T>>::remove_prefix(cc, None);
+		<EndorseeCount<T>>::insert(cc, 0);
 
-			<NewbieRegistry<T>>::remove_prefix((cid, cindex), None);
-			<NewbieIndex<T>>::remove_prefix((cid, cindex), None);
-			<NewbieCount<T>>::insert((cid, cindex), 0);
+		<NewbieRegistry<T>>::remove_prefix(cc, None);
+		<NewbieIndex<T>>::remove_prefix(cc, None);
+		<NewbieCount<T>>::insert(cc, 0);
 
-			<AssignmentCounts<T>>::insert((cid, cindex), AssignmentCount::default());
+		<AssignmentCounts<T>>::insert(cc, AssignmentCount::default());
 
-			Assignments::<T>::remove((cid, cindex));
+		Assignments::<T>::remove(cc);
 
-			<Endorsees<T>>::remove_prefix((cid, cindex), None);
-			<MeetupCount<T>>::insert((cid, cindex), 0);
-			<AttestationRegistry<T>>::remove_prefix((cid, cindex), None);
-			<AttestationIndex<T>>::remove_prefix((cid, cindex), None);
-			<AttestationCount<T>>::insert((cid, cindex), 0);
-			<MeetupParticipantCountVote<T>>::remove_prefix((cid, cindex), None);
+		<Endorsees<T>>::remove_prefix(cc, None);
+		<MeetupCount<T>>::insert(cc, 0);
+		<AttestationRegistry<T>>::remove_prefix(cc, None);
+		<AttestationIndex<T>>::remove_prefix(cc, None);
+		<AttestationCount<T>>::insert(cc, 0);
+		<MeetupParticipantCountVote<T>>::remove_prefix(cc, None);
 
-			<IssuedRewards<T>>::remove_prefix((cid, cindex), None);
+		<IssuedRewards<T>>::remove_prefix(cc, None);
+	}
+
+	fn purge_registry(cindex: CeremonyIndexType) {
+		let cids = <encointer_communities::Pallet<T>>::community_identifiers();
+		for cid in cids.iter() {
+			Self::purge_community_ceremony((*cid, cindex));
 		}
```

Comment:
Even though, copy-dereferencing is already cheap, you can save a bit of efficiency by using `into_iter`, which returns owned values.